gpt_index/indices/query/tree/leaf_query.py

```diff
@@ -125,7 +125,13 @@ def _query_level(
         query_str = query_bundle.query_str
         cur_node_list = get_sorted_node_list(cur_nodes)
 
-        if self.child_branch_factor == 1:
+        if len(cur_node_list) == 1:
+            logging.debug(f">[Level {level}] Only one node left. Querying node.")
+            result_response = self._query_with_selected_node(
```

Comment:
@kpister i'd return this function call directly to avoid linting errors with the `result_response` variable below.

(also no need to cast to string below)

Reply:
Updated